### ISSUE TITLE
train: Stage B intent fine-tune run (50k steps, intent-balanced)

### ISSUE THREAD

**Issue by AbdelStark** [AbdelStark/CodingJEPA#76]
## Context

RFC-0008 §D2 Stage B + §D8: load Stage A checkpoint, reset optimizer + LR warmup, intent-balanced sampling on labeled pairs. Effective batch 512 pairs.

## Scope

- Launch `python -m codingjepa train finetune` against `config/train/finetune.yaml` initialized from `runs/pretrain-v1/checkpoints/best.ckpt`.
- Output: `runs/finetune-v1/checkpoints/best.ckpt`.

## Out of Scope

Pretrain (separate issue).

## Design Reference

- RFC: [RFC-0008 §D2 §D8](../blob/main/docs/rfcs/RFC-0008-training-recipe-and-pipeline.md)
- SPEC: [`docs/spec/08-performance-budget.md`](../blob/main/docs/spec/08-performance-budget.md)

## Acceptance Criteria

- [ ] `runs/finetune-v1/checkpoints/best.ckpt` exists.
- [ ] Wall-clock ≤ 24 h.
- [ ] Rank and loss-monotonicity gates pass.

## Dependencies

- Depends on: #75, T-DATA pairs frozen.

## Metadata

```
type: feature
area: training
priority: p0
effort: l
spec: rfc-0008
milestone: v0.3
tracking: #6
```
